app/assets/javascripts/components/categories/add_category_button.jsx

```diff
@@ -1,132 +1,122 @@
 import { resetScopingMethod } from '../../actions/scoping_methods';
 
-const AddCategoryButton = createReactClass({
-  displayName: 'AddCategoryButton',
-
-  propTypes: {
-    course: PropTypes.object.isRequired,
-    is_open: PropTypes.bool,
-    open: PropTypes.func.isRequired,
-    initiateConfirm: PropTypes.func,
-    addCategory: PropTypes.func,
-    source: PropTypes.string.isRequired
-  },
+const AddCategoryButton = ({
+  course,
+  source,
+  addCategory,
+}) => {
+  const scopingMethods = useSelector(state => state.scopingMethods);
+  const dispatch = useDispatch();
 
-  getKey() {
-    return `add_${this.props.source}_button`;
-  },
+  const getKey = () => `add_${source}_button`;
+  const { isOpen, ref, open } = useExpandablePopover(getKey);
 
-  addCategory(e) {
+  const addCategoryHandler = (e) => {
     e.preventDefault();
-    const addCategory = this.props.addCategory;
     const payload = {
       ...getAddCategoriesPayload({
-        sourceType: this.props.source,
-        scopingMethods: this.props.scopingMethods,
+        sourceType: source,
+        scopingMethods,
       }),
-      course_id: this.props.course.id,
-      source: this.props.source,
+      course_id: course.id,
+      source,
     };
-    const resetScoping = this.props.resetScopingMethod;
-    const open = this.props.open;
-    const onConfirm = function () {
+    const onConfirm = () => {
       addCategory(payload);
-      resetScoping();
+      dispatch(resetScopingMethod());
       open(null);
     };
     const confirmMessage = I18n.t('categories.confirm_add_n_scoping_method', { n: payload.categories.items.length });
-    return this.props.initiateConfirm({ confirmMessage, onConfirm });
-  },
+    dispatch(initiateConfirm({ confirmMessage, onConfirm }));
+  };
 
+  let description;
 
-  render() {
-    let description;
-      if (this.props.source === 'pileid') {
-        description = (
-          <p>
-            Make sure the PagePile&apos;s wiki is one of the tracked wikis for this program.
-          </p>
-        );
-      } else if (this.props.source === 'psid') {
-        description = (
-          <p>
-            PetScan queries only work with the Dashboard one wiki at a time. Ensure that your PetScan query includes only articles from a single wiki, and that it matches the wiki set below.
-          </p>
-        );
-      }
-    const permitted = true;
-    let className = 'button border small assign-button';
-    if (this.props.is_open) { className += ' dark'; }
+  if (source === 'pileid') {
+    description = (
+      <p>
+        {I18n.t('categories.pileid_description')}
+      </p>
+    );
+  } else if (source === 'psid') {
+    description = (
+      <p>
+        {I18n.t('categories.psid_description')}
+      </p>
+    );
+  }
 
-    const buttonText = I18n.t(`categories.add_${this.props.source}`);
-    const showButton = <button className={`${className}`} onClick={this.props.open}>{buttonText}</button>;
+  const permitted = true;
```

Comment:
`permitted` is always `true` and is not reassigned. This means the following `if (permitted)` always executes. This variable and the ensuing if block nesting should either be removed or check if the `permitted` needs to be modified based on something rather than always being `true`.